domain/modelconfig/validators/validators.go

```diff
@@ -77,6 +77,22 @@ func AuthorizedKeysChange() config.ValidatorFunc {
 	}
 }
 
+// ContainerNetworkingMethodChange checks to see if there has been any change
+// to a model config's container networking method.
+func ContainerNetworkingMethodChange() config.ValidatorFunc {
+	return func(ctx context.Context, cfg, old *config.Config) (*config.Config, error) {
+		if cfg.ContainerNetworkingMethod() == old.ContainerNetworkingMethod() {
+			// No change. Nothing more todo.
+			return cfg, nil
+		}
+
+		return cfg, &config.ValidationError{
+			InvalidAttrs: []string{config.ContainerNetworkingMethod},
+			Reason:       "container-networking-method cannot be changed",
```

Comment:
Great question. We discussed this in the APAC team and we agree that this might be a bug in the current implementation.

Currently we only run logic to translate the value supplied by the user at model creation or bootstrap time. As we believe that providers don't change and we don't understand the implications of what would happen if container networking suddenly change over time we think we should opt on the side of set once.

Reply:
We also don't do any validation on what the value is when set by the user.